component/otelcol/receiver/jaeger/jaeger.go

```diff
@@ -0,0 +1,205 @@
+	if args.Protocols.GRPC == nil &&
+		args.Protocols.ThriftHTTP == nil &&
+		args.Protocols.ThriftBinary == nil &&
+		args.Protocols.ThriftCompact == nil {
+
+		return fmt.Errorf("at least one protocol must be enabled")
+	}
+
+	return nil
+}
+
+// Convert implements receiver.Arguments.
+func (args Arguments) Convert() otelconfig.Receiver {
+	return &jaegerreceiver.Config{
+		ReceiverSettings: otelconfig.NewReceiverSettings(otelconfig.NewComponentID("jaeger")),
+		Protocols: jaegerreceiver.Protocols{
+			GRPC:          args.Protocols.GRPC.Convert(),
+			ThriftHTTP:    args.Protocols.ThriftHTTP.Convert(),
+			ThriftBinary:  args.Protocols.ThriftBinary.Convert(),
+			ThriftCompact: args.Protocols.ThriftCompact.Convert(),
+		},
+		RemoteSampling: args.RemoteSampling.Convert(),
+	}
+}
+
+// Extensions implements receiver.Arguments.
+func (args Arguments) Extensions() map[otelconfig.ComponentID]otelcomponent.Extension {
+	if args.RemoteSampling == nil {
+		return nil
+	}
+	return args.RemoteSampling.Client.Extensions()
+}
+
+// Exporters implements receiver.Arguments.
+func (args Arguments) Exporters() map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter {
+	return nil
+}
+
+// NextConsumers implements receiver.Arguments.
+func (args Arguments) NextConsumers() *otelcol.ConsumerArguments {
+	return args.Output
+}
+
+// ProtocolsArguments configures protocols for otelcol.receiver.jaeger to
+// listen on.
+type ProtocolsArguments struct {
+	GRPC          *otelcol.GRPCServerArguments `river:"grpc,block,optional"`
+	ThriftHTTP    *otelcol.HTTPServerArguments `river:"thrift_http,block,optional"`
+	ThriftBinary  *ProtocolUDP                 `river:"thrift_binary,block,optional"`
+	ThriftCompact *ProtocolUDP                 `river:"thrift_compact,block,optional"`
+}
+
+// ProtocolUDP configures a UDP server.
+type ProtocolUDP struct {
+	Endpoint         string           `river:"endpoint,attr,optional"`
+	QueueSize        int              `river:"queue_size,attr,optional"`
+	MaxPacketSize    units.Base2Bytes `river:"max_packet_size,attr,optional"`
+	Workers          int              `river:"workers,attr,optional"`
+	SocketBufferSize units.Base2Bytes `river:"socket_buffer_size,attr,optional"`
+}
+
+// Convert converts proto into the upstream type.
+func (proto *ProtocolUDP) Convert() *jaegerreceiver.ProtocolUDP {
+	if proto == nil {
+		return nil
+	}
+
+	return &jaegerreceiver.ProtocolUDP{
+		Endpoint: proto.Endpoint,
+		ServerConfigUDP: jaegerreceiver.ServerConfigUDP{
+			QueueSize:        proto.QueueSize,
+			MaxPacketSize:    int(proto.MaxPacketSize),
+			Workers:          proto.Workers,
+			SocketBufferSize: int(proto.SocketBufferSize),
+		},
+	}
+}
+
+// RemoteSamplingArguments configures remote sampling settings.
+type RemoteSamplingArguments struct {
+	HostEndpoint               string                      `river:"host_endpoint,attr"`
+	StrategyFile               string                      `river:"strategy_file,attr"`
```

Comment:
StrategyFile feels backwards in a world where we have local.file but I dont see a super easy way around it. I think longterm this can represent something maybe we could upstream to take in an optional func or somesuch,

Reply:
Yeah, agreed, I'll add a TODO 👍